infra/src/main/scala/ch/epfl/scala/services/github/GithubClient.scala

```diff
@@ -184,24 +220,117 @@ class GithubClient(githubConfig: GithubConfig)(implicit system: ActorSystem) ext
     }
   }
 
+  def fetchMyRepo(myToken: Secret): Future[Map[GithubRepo, String]] = {
```

Comment:
```suggestion
  def fetchUserRepos(userToken: Secret): Future[Map[GithubRepo, String]] = {
```

Reply:
Also I would pass the requested permission as a list and filter them at the client side. Otherwise it is not clear that `String` in `Map[GithubRepo, String]` is a permission.
```suggestion
  def fetchMyRepo(myToken: Secret, permissions: Seq[String]): Future[Seq[GithubRepo]] = {
```

Reply:
yep!